implementations/rust/ockam/ockam_node/src/context/send_message.rs

```diff
@@ -149,18 +210,97 @@ impl Context {
-            if let Some(incoming_access_control) = options.incoming_access_control {
-                incoming_access_control
+    #[allow(clippy::too_many_arguments)]
+    pub(super) async fn make_send_context(
+        runtime: Handle,
+        router: Arc<Router>,
+        transports: Arc<RwLock<HashMap<TransportType, Arc<dyn Transport>>>>,
+        flow_controls: &FlowControls,
+        mailbox_count: Arc<AtomicUsize>,
+        next: Address,
+        outgoing_access_control: Option<Arc<dyn OutgoingAccessControl>>,
+        #[cfg(feature = "std")] tracing_context: OpenTelemetryContext,
+    ) -> Result<Context> {
+        let address = Address::random_tagged("Context.send.detached");
+
+        let outgoing_access_control: Arc<dyn OutgoingAccessControl> =
+            if let Some(outgoing_access_control) = outgoing_access_control {
+                Arc::new(AllOutgoingAccessControl::new(vec![
+                    outgoing_access_control,
+                    Arc::new(AllowOnwardAddress(next.clone())),
+                ]))
             } else {
-                Arc::new(AllowAll)
+                Arc::new(AllowOnwardAddress(next.clone()))
             };
 
+        let mailboxes = Mailboxes::new(
+            Mailbox::new(
+                address.clone(),
+                None,
+                Arc::new(AllowAll),
+                outgoing_access_control,
+            ),
+            vec![],
+        );
+
+        let child_ctx = Self::new_detached_with_mailboxes_impl(
+            runtime,
+            router,
+            transports,
+            flow_controls,
+            mailboxes,
+            mailbox_count,
+        )?;
+
+        #[cfg(feature = "std")]
+        child_ctx.set_tracing_context(tracing_context);
+        Ok(child_ctx)
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub(super) async fn make_send_and_receive_context(
+        runtime: Handle,
+        router: Arc<Router>,
+        transports: Arc<RwLock<HashMap<TransportType, Arc<dyn Transport>>>>,
+        flow_controls: &FlowControls,
+        mailbox_count: Arc<AtomicUsize>,
+        next: Address,
+        incoming_access_control: Option<Arc<dyn IncomingAccessControl>>,
+        outgoing_access_control: Option<Arc<dyn OutgoingAccessControl>>,
+        #[cfg(feature = "std")] tracing_context: OpenTelemetryContext,
+    ) -> Result<Context> {
+        let address = Address::random_tagged("Context.send_and_receive.detached");
+
+        let incoming_access_control = if let Some(incoming_access_control) = incoming_access_control
+        {
+            incoming_access_control
+        } else {
+            Arc::new(AllowAll)
+        };
+
         let outgoing_access_control: Arc<dyn OutgoingAccessControl> =
-            if let Some(outgoing_access_control) = options.outgoing_access_control {
+            if let Some(outgoing_access_control) = outgoing_access_control {
                 Arc::new(AllOutgoingAccessControl::new(vec![
```

Comment:
I know this is not related to your change, but I think `AllowOnwardAddress` is not needed in this case. Both access controls can be `incoming/outgoing_access_control.unwrap_or(AllowAll)`. Would you mind making that change? Or I can do it in a separate PR